File: Snippets/Bin_Dec/bd.r
#
# Take a binary representation, return its decimal equivalent
#
bin2dec <- function (bin) {
    return (strtoi (bin, 2))
}


#
# Given a decimal number, return its binary representation
#
dec2bin <- function (dec) {
    return (as.integer (paste (as.integer (rev (intToBits (dec))),
                               collapse = "")))
}

File: Snippets/Bin_Dec/bd.py
#
# Take a binary representation, return its decimal equivalent
#
def bin2dec (bin):
    return (int (bin, 2))


#
# Given a decimal number, return its binary representation
#
def dec2bin (dec):
    return (bin (dec) [2:])



for i in range (16):
    mybin = dec2bin (i)
    mydec = bin2dec (mybin)
    print (i, " ", mybin, " ", mydec)
